### PR TITLE
Add option for disabling natural sheep spawns

### PR DESCRIPTION
**Pull request by tali713**
What it says on the tin, partial implementation of #7.

### CODE REVIEW

**Comment by josephcsible** [comment]
In the future, you can fix pull requests by just pushing over the same branch, rather than having to make a new one.

**Comment by tali713** [comment]
Normally that is what I do, however, I just moved to a new computer and forgot to set my email address, and that's per commit,  I was able to get one of the commits address changed and then gave up.  Sorry.  This issue won't come up again until I forget on my next machine.

Ammending commits just isn't something I normally do.

**Comment by rubensworks** [comment]
I was thinking of just adding a spawn rate option. When the value would be set to `0`, spawning would automatically be disabled.

So I'd rather go in that direction, to avoid having to remove and add another config option in the future. Feel free to adapt this PR to do so, otherwise I'll try to do it in the future :-)

**Comment by josephcsible** [comment]
Even in that case you didn't have to do this; you can amend commits and force push.

**Comment by tali713** [comment]
Something like this?

File: src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java [hunk 1]
```diff
@@ -107,7 +115,7 @@ public int getForegroundEggColor() {
     public void onRegistered() {
         super.onRegistered();
         for (Biome biome : Biome.REGISTRY) {
```
Comment (rubensworks) [inline src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java]: Make sure to skip the loop if `spawnChance <= 0`. No need to iterate all biomes when the weight is zero in any case.

File: src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java [hunk 2]
```diff
@@ -107,7 +115,7 @@ public int getForegroundEggColor() {
...
-            EntityRegistry.addSpawn(EntityEnergeticSheep.class, 5, 2, 4, EnumCreatureType.CREATURE, biome);
+            EntityRegistry.addSpawn(EntityEnergeticSheep.class, spawnChance, 2, 4, EnumCreatureType.CREATURE, biome);
```
Comment (rubensworks) [inline src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java]: This is actually not a chance, but a spawn weight (values have no upper limit). So I would rename this option (and description accordingly).

**Review by rubensworks — CHANGES_REQUESTED** [review]

File: src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java (hunk 1 above)
Comment (tali713) [reply src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java]: Done

File: src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java (hunk 2 above)
Comment (tali713) [reply src/main/java/org/cyclops/energeticsheep/entity/EntityEnergeticSheepConfig.java]: I am finding it rather difficult to concisely describe what this option does, without explaining spawning mechanics.

**Review by rubensworks — APPROVED** [review]

**Comment by rubensworks** [comment]
Looks good, thanks!

**Pull request merged**